### PR TITLE
MAKE-387 Clean up download tests and require absolute paths for downloads

### PR DESCRIPTION
**Pull request by Kimchelly**
* Also move download-related functions from rest_service.go to download.go
* More tests for download

### CODE REVIEW

File: download.go [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+		if catcher.HasErrors() {
+			errOut <- catcher.Resolve()
+		}
+		close(errOut)
+	}()
+
+	return output, errOut
+}
+
+func aggregateErrors(groups ...<-chan error) error {
```
Comment (tychoish) [inline download.go]: you can just pass catcher around, they're threadsafe.

File: download.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+	return nil
+}
+
+func addMongoDBFilesToCache(cache *lru.Cache, absRootPath string) func(string) error {
+	return func(fileName string) error {
+		catcher := grip.NewBasicCatcher()
+
+		filePath := filepath.Join(absRootPath, fileName)
+		if err := cache.AddFile(filePath); err != nil {
+			catcher.Add(errors.Wrapf(err, "problem adding file %s to cache", filePath))
```
Comment (tychoish) [inline download.go]: should return rather than catch things.

File: download.go [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+		return errors.Wrap(err, "problem starting download job queue")
+	}
+
+	if err := amboy.PopulateQueue(ctx, q, jobs); err != nil {
+		return errors.Wrap(err, "problem adding download jobs to queue")
+	}
+
+	go func() {
+		if err := processJobs(q); err != nil {
+			grip.Errorf(errors.Wrap(err, "error occurred while adding jobs to cache").Error())
```
Comment (tychoish) [inline download.go]: what do we do this this error?

File: jrpc/internal/service.go [hunk 4]
```diff
@@ -232,12 +232,14 @@ func (s *jasperService) ResetTags(ctx context.Context, id *JasperProcessID) (*Op
 
 func (s *jasperService) DownloadMongoDB(ctx context.Context, opts *MongoDBDownloadOptions) (*OperationOutcome, error) {
 	jopts := opts.Export()
-	if err := jopts.BuildOpts.Validate(); err != nil {
-		return &OperationOutcome{Success: false, Text: errors.Wrap(err, "invalid build options").Error()}, err
+	if err := jopts.Validate(); err != nil {
+		err = errors.Wrap(err, "problem validating MongoDB download options")
+		return &OperationOutcome{Success: false, Text: err.Error()}, err
```
Comment (tychoish) [inline jrpc/internal/service.go]: I would just combine this here and below into one longer line,

File: download.go (hunk 2 above)
Comment (Kimchelly) [reply download.go]: Done.

File: download.go (hunk 1 above)
Comment (Kimchelly) [reply download.go]: Changed to use catcher.

File: jrpc/internal/service.go (hunk 4 above)
Comment (Kimchelly) [reply jrpc/internal/service.go]: Done. This is done in several other places. Do you want them all changed like this?

File: download.go (hunk 3 above)
Comment (Kimchelly) [reply download.go]: Truthfully, not much. It's only there to print a message in case an error occurs while it's asynchronously downloading.

**Pull request merged**